### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ChungYiTsai**

### CODE REVIEW

File: app/application/controllers/app.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            target_made = Service::AddTarget.new.call(target_request)
+
+            if target_made.failure?
+              flash[:error] = target_made.failure
+              routing.redirect '/'
+            end
+
+            target = target_made.value!
+            session[:watching].insert(0, target.company_name).uniq!
+            #f lash[:notice] = 'target added to your list'
```
Comment (tienyulin) [inline app/application/controllers/app.rb:63]: Please remove this commented-out code

File: app/application/forms/new_project.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+      rule(:company_name) do
+        if check_valid(value)
+          key.failure('is a not define company.')
+        end
+      end
+
+      def check_valid(company)
+        COMPANY_LIST[0][company.downcase].nil?
+      end
```
Comment (tienyulin) [inline app/application/forms/new_project.rb:24]: if `check_valid` only use for `rule(:company_name)`, I would suggest just putting the logic in `rule(:company_name)`. No need to create a method.

File: app/application/services/add_target.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            Success(input)
+        rescue StandardError => error
+            Failure(error.to_s)
+        end
+
+        def store_target(input)
+            target =
+                if (new_target = input[:remote_target])
+                    Repository::For.entity(new_target).create(new_target)
+                elsif(update_target = input[:update_target])
```
Comment (tienyulin) [inline app/application/services/add_target.rb:47]: Wrong operator for equal, equal is `==` not `=`

**Review by tienyulin — COMMENTED** [review]
Reviewed

File: app/application/controllers/app.rb (hunk 1 above)
Comment (ChungYiTsai) [reply app/application/controllers/app.rb:63]: OK

**Pull request merged**